Objective-C/CBLReplicator.mm

```diff
@@ -358,6 +358,67 @@ - (void) removeChangeListenerWithToken: (id<CBLListenerToken>)token {
     }
 }
 
+- (NSSet<NSString*>*) pendingDocumentIds: (NSError**)error {
+    if (_config.replicatorType > 1) {
+        if (error)
+            *error = [NSError errorWithDomain: CBLErrorDomain
+                                         code: CBLErrorUnsupported
+                                     userInfo: @{NSLocalizedDescriptionKey: kCBLErrorMessagePullOnlyPendingDocIDs}];
+        return nil;
+    }
+    
+    C4Error c4err = {};
+    if (![self _setupC4Replicator: &c4err]) {
+        convertError(c4err, error);
+        CBLWarnError(Sync, @"%@: Replicator cannot be created: %d/%d", self, c4err.domain, c4err.code);
+        return nil;
+    }
+    
+    C4SliceResult result = c4repl_getPendingDocIDs(_repl, &c4err);
+    if (c4err.code > 0) {
+        convertError(c4err, error);
+        CBLWarnError(Sync, @"Error while fetching pending documentIds: %d/%d", c4err.domain, c4err.code);
+        return nil;
+    }
+
+    if (result.size <= 0)
+        return [NSSet set];
+
+    FLValue val = FLValue_FromData(C4Slice(result), kFLTrusted);
+    NSArray<NSString*>* list = FLValue_GetNSObject(val, nullptr);
+
+    return [NSSet setWithArray: list];
+}
+
+- (BOOL) isDocumentPending: (NSString*)documentID error: (NSError**)error {
+    CBLAssertNotNil(documentID);
+
+    if (_config.replicatorType > 1) {
+        if (error)
+            *error = [NSError errorWithDomain: CBLErrorDomain
+                code: CBLErrorUnsupported
+            userInfo: @{NSLocalizedDescriptionKey: kCBLErrorMessagePullOnlyPendingDocIDs}];
+        return NO;
+    }
+
+    C4Error c4err = {};
+    if (![self _setupC4Replicator: &c4err]) {
```

Comment:
Should be called under the same lock as the lock in the start method for thread safety.

Reply:
The Java code is nearly identical.
The reference to the C4Replicator is guarded by a lock, because it could be mutated on multiple threads.  Once the code has a non-null reference, though, it presumes it can used it freely.  If that is not the case, there will have to be a *LOT* more locking in the Java code.

Reply:
Yes, it's the same concept here. Basically we need to guard the C4Replicator so that it will not create multiple C4Replicator if it got called from multiple threads.